xmodule/video_block/video_handlers.py

```diff
@@ -562,22 +574,33 @@ def studio_transcript(self, request, dispatch):
                 if edx_video_id:
                     delete_video_transcript(video_id=edx_video_id, language_code=language)
 
-                if language == 'en':
-                    # remove any transcript file from content store for the video ids
-                    possible_sub_ids = [
-                        self.sub,  # pylint: disable=access-member-before-definition
-                        self.youtube_id_1_0
-                    ] + get_html5_ids(self.html5_sources)
-                    for sub_id in possible_sub_ids:
-                        remove_subs_from_store(sub_id, self, language)
-
-                    # update metadata as `en` can also be present in `transcripts` field
-                    remove_subs_from_store(self.transcripts.pop(language, None), self, language)
-
-                    # also empty `sub` field
-                    self.sub = ''  # pylint: disable=attribute-defined-outside-init
+                if isinstance(self.scope_ids.usage_id.context_key, LibraryLocatorV2):
+                    transcript_file_path = f"static/{self.transcripts.pop(language, None)}"
+                    lib_api.delete_library_block_static_asset_file(self.scope_ids.usage_id, transcript_file_path)
```

Comment:
If `transcript_file_path == "static/None"`, I think `delete_library_block_static_asset_file` raises a validation error. By contrast, the `remove_subs_from_store` method just passes if the transcript file isn't found.

So maybe we should check for that here (at the risk of enraging pylint further..):
```suggestion
                    transcript_path = self.transcripts.pop(language, None)
                    if transcript_path:
                        lib_api.delete_library_block_static_asset_file(self.scope_ids.usage_id, f"static/{transcript_path}")
```

Reply:
Updated https://github.com/openedx/edx-platform/pull/36058/commits/4532a2ae41af434503964bcde762c8e196936de5